compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1784,7 +1785,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     Some(trait_ref) => trait_ref,
                     None => {
                         // A cycle error occurred, most likely.
-                        return Err(ErrorReported);
+                        return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
This should be a `delay_span_bug`, I think? I believe `impl_trait_ref` returning `Option` has a use for `None` that isn't an error (otherwise I would suggest `Result`-ifying it).